src/integrations/iron_handlers.rs

```diff
@@ -77,59 +85,77 @@ impl<'a, CtxFactory, Query, Mutation, CtxT>
     }
 
     fn handle_post(&self, req: &mut Request) -> IronResult<Response> {
-        let json_data = itry!(Json::from_reader(&mut req.body));
-
-        let json_obj = match json_data {
-            Json::Object(o) => o,
-            _ => return Ok(Response::with((status::BadRequest, "No JSON object was decoded"))),
-        };
-
-        let mut query = None;
-        let mut variables = Variables::new();
-
-        for (k, v) in json_obj {
-            if k == "query" {
-                query = v.as_string().map(|s| s.to_owned());
+        let mut request_payload = String::new();
+        itry!(req.body.read_to_string(&mut request_payload));
+        let json_data =
+            match serde_json::from_str::<Json>(&*request_payload) {
+                Ok(json) => json,
+                Err(err) => {
+                    let error = IronError::new(
+                        Box::new(GraphQlIronError::Serde(err)),
+                        (status::BadRequest, "No JSON object was decoded."));
+                    return Err(error)
+                }
+            };
+        match json_data {
+            Json::Object(json_obj) => {
+                let mut query = None;
+                let mut variables = Variables::new();
+
+                for (k, v) in json_obj {
+                    if k == "query" {
+                        query = serde_json::to_string(&v)
+                            .ok()
+                            .map(|query| query.replace("\"", ""));
```

Comment:
This sounds like a case of double encoding - what happens if you send `{droid(id: "1001"){name}}`?